subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstyleExtension.java

```diff
@@ -37,10 +39,11 @@
     private int maxErrors;
     private int maxWarnings = Integer.MAX_VALUE;
     private boolean showViolations = true;
-    private File configDir;
+    private Property<File> configDir;
 
     public CheckstyleExtension(Project project) {
         this.project = project;
+        configDir = project.getObjects().property(File.class);
```

Comment:
❌ Should be a [directory property](https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java#L64).

Reply:
I think at the moment there's not a way to set the value of a `DirectoryProperty` lazily e.g. with a `Callable`. The `set` method just takes a `File` instance. In turn that would render the deprecation message again even if we don't need it. Is there a way to defer the evaluation of the value with `DirectoryProperty`?

Reply:
It`s a `Property<Directory>`, so you can use a `Provider<Directory>`.

Reply:
Done though I feel like the code looks more complex now.